File: formbuilder-backend1/src/main/java/com/sttl/formbuilder2/util/ApiConstants.java
package com.sttl.formbuilder2.util;

/**
 * Centralized API Path Constants
 * 
 * Purpose: Single source of truth for all API endpoint paths and versions.
 * When API version changes, update VERSION constant and all paths automatically update.
 * 
 * Usage:
 *   @RequestMapping(ApiConstants.AUTH_BASE)
 *   @GetMapping(ApiConstants.AUTH_ME)
 */
public final class ApiConstants {
    
    private ApiConstants() {
        // Private constructor to prevent instantiation
        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
    
    // ═══════════════════════════════════════════════════════════════
    // VERSION MANAGEMENT
    // ═══════════════════════════════════════════════════════════════
    
    /** Current API Version - Change this to update all endpoints */
    public static final String VERSION = "v1";
    
    /** Base API Path */
    public static final String API_BASE = "/api/" + VERSION;
    
    // ═══════════════════════════════════════════════════════════════
    // AUTHENTICATION & SESSION
    // ═══════════════════════════════════════════════════════════════
    
    public static final String AUTH_BASE = API_BASE + "/auth";
    public static final String AUTH_LOGIN = "/login";
    public static final String AUTH_LOGOUT = "/logout";
    public static final String AUTH_REGISTER = "/register";
    public static final String AUTH_ME = "/me";
    public static final String AUTH_PERMISSIONS = "/permissions";
    
    // ═══════════════════════════════════════════════════════════════
    // FORM MANAGEMENT
    // ═══════════════════════════════════════════════════════════════
    
    public static final String FORMS_BASE = API_BASE + "/forms";
    public static final String FORMS_LIST = "";
    public static final String FORMS_CREATE = "";
    public static final String FORMS_GET = "/{id}";
    public static final String FORMS_UPDATE = "/{id}";
    public static final String FORMS_DELETE = "/{id}";
    public static final String FORMS_RESTORE = "/{id}/restore";
    public static final String FORMS_DELETE_PERMANENT = "/{id}/permanent";
    public static final String FORMS_ARCHIVED = "/archived";
    public static final String FORMS_STATS = "/stats";
    
    // Public Form Access (No Auth Required)
    public static final String FORMS_PUBLIC = "/public/{token}";
    public static final String FORMS_PUBLIC_SUBMISSIONS = "/public/{token}/submissions";
    public static final String FORMS_PUBLIC_SUBMISSION_GET = "/public/{token}/submissions/{submissionId}";
    public static final String FORMS_PUBLIC_SUBMISSION_UPDATE = "/public/{token}/submissions/{submissionId}";
    
    // ═══════════════════════════════════════════════════════════════
    // FORM VERSIONS
    // ═══════════════════════════════════════════════════════════════
    
    public static final String VERSIONS_BASE = API_BASE + "/forms/{formId}/versions";
    public static final String VERSIONS_CREATE = "";
    public static final String VERSIONS_LIST = "";
    public static final String VERSIONS_GET = "/{versionId}";
    public static final String VERSIONS_ACTIVATE = "/{versionId}/activate";
    
    // ═══════════════════════════════════════════════════════════════
    // SUBMISSIONS
    // ═══════════════════════════════════════════════════════════════
    
    public static final String SUBMISSIONS_BASE = "/{id}/submissions";
    public static final String SUBMISSIONS_LIST = "/{id}/submissions";
    public static final String SUBMISSIONS_CREATE = "/{id}/submissions";
    public static final String SUBMISSIONS_GET = "/{formId}/submissions/{submissionId}";
    public static final String SUBMISSIONS_UPDATE = "/{formId}/submissions/{submissionId}";
    public static final String SUBMISSIONS_DELETE = "/{formId}/submissions/{submissionId}";
    public static final String SUBMISSIONS_BULK = "/{formId}/submissions/bulk";
    public static final String SUBMISSIONS_BULK_DELETE = "/{formId}/submissions/bulk";
    public static final String SUBMISSIONS_RESTORE = "/{id}/submissions/{submissionId}/restore";
    public static final String SUBMISSIONS_EXPORT = "/{id}/submissions/export";
    
    // Lookup Values
    public static final String SUBMISSIONS_LOOKUP_VALUES = "/{id}/columns/{fieldKey}/values";
    
    // ═══════════════════════════════════════════════════════════════
    // RUNTIME (Form Fill)
    // ═══════════════════════════════════════════════════════════════
    
    public static final String RUNTIME_BASE = API_BASE + "/runtime";
    public static final String RUNTIME_GET_FORM = "/forms/{formCode}";
    public static final String RUNTIME_SUBMIT = "/forms/{formCode}/submissions";
    public static final String RUNTIME_SAVE_DRAFT = "/forms/{formCode}/drafts";
    
    // ═══════════════════════════════════════════════════════════════
    // WORKFLOW (Approval System)
    // ═══════════════════════════════════════════════════════════════
    
    public static final String WORKFLOW_BASE = API_BASE + "/workflows";
    public static final String WORKFLOW_INITIATE = "/initiate";
    public static final String WORKFLOW_APPROVE = "/steps/{id}/approve";
    public static final String WORKFLOW_REJECT = "/steps/{id}/reject";
    public static final String WORKFLOW_AUTHORITIES = "/available-authorities";
    public static final String WORKFLOW_MY_SUBMISSIONS = "/my-submissions";
    public static final String WORKFLOW_MY_PENDING = "/my-pending";
    public static final String WORKFLOW_MY_HANDLED = "/my-handled";
    public static final String WORKFLOW_FIX_DB = "/fix-db";
    
    // ═══════════════════════════════════════════════════════════════
    // ADMIN - USER MANAGEMENT
    // ═══════════════════════════════════════════════════════════════
    
    public static final String ADMIN_USERS_BASE = API_BASE + "/admin/users";
    public static final String ADMIN_USERS_LIST = "";
    public static final String ADMIN_USERS_CREATE = "";
    public static final String ADMIN_USERS_GET = "/{id}";
    public static final String ADMIN_USERS_UPDATE = "/{id}";
    public static final String ADMIN_USERS_DELETE = "/{id}";
    public static final String ADMIN_USERS_TOGGLE_DELETE = "/{id}/toggle-delete";
    public static final String ADMIN_USERS_ASSIGN_ROLE = "/{userId}/roles";
    public static final String ADMIN_USERS_REVOKE_ROLE = "/{userId}/roles/{assignmentId}";
    
    // ═══════════════════════════════════════════════════════════════
    // ADMIN - ROLE MANAGEMENT
    // ═══════════════════════════════════════════════════════════════
    
    public static final String ADMIN_ROLES_BASE = API_BASE + "/admin/roles";
    public static final String ADMIN_ROLES_LIST = "";
    public static final String ADMIN_ROLES_CREATE = "";
    public static final String ADMIN_ROLES_GET = "/{id}";
    public static final String ADMIN_ROLES_UPDATE = "/{id}";
    public static final String ADMIN_ROLES_DELETE = "/{id}";
    
    // ═══════════════════════════════════════════════════════════════
    // ADMIN - PERMISSIONS
    // ═══════════════════════════════════════════════════════════════
    
    public static final String ADMIN_PERMISSIONS_BASE = API_BASE + "/admin/permissions";
    public static final String ADMIN_PERMISSIONS_LIST = "";
    
    // ═══════════════════════════════════════════════════════════════
    // ADMIN - AUDIT LOGS
    // ═══════════════════════════════════════════════════════════════
    
    public static final String ADMIN_AUDIT_BASE = API_BASE + "/admin/audit";
    public static final String ADMIN_AUDIT_LIST = "";
    
    // ═══════════════════════════════════════════════════════════════
    // ADMIN - APPROVAL HISTORY
    // ═══════════════════════════════════════════════════════════════
    
    public static final String ADMIN_APPROVALS_BASE = API_BASE + "/admin/approvals/history";
    public static final String ADMIN_APPROVALS_LIST = "";
    
    // ═══════════════════════════════════════════════════════════════
    // USER PROFILE
    // ═══════════════════════════════════════════════════════════════
    
    public static final String PROFILE_BASE = API_BASE + "/profile";
    public static final String PROFILE_UPDATE = "";
    public static final String PROFILE_CHANGE_PASSWORD = "/change-password";
    
    // ═══════════════════════════════════════════════════════════════
    // MODULES & MENU (Dynamic Navigation)
    // ═══════════════════════════════════════════════════════════════
    
    public static final String MODULES_BASE = API_BASE + "/modules";
    public static final String MODULES_LIST = "";
    
    public static final String MENU_BASE = API_BASE + "/menu";
    public static final String MENU_LIST = "";
    
    // ═══════════════════════════════════════════════════════════════
    // ROLE-MODULE MAPPING
    // ═══════════════════════════════════════════════════════════════
    
    public static final String ROLE_MODULES_BASE = API_BASE + "/roles";
    public static final String ROLE_MODULES_GET = "/{roleId}/modules";
    public static final String ROLE_MODULES_ASSIGN = "/{roleId}/modules";
    public static final String ROLE_MODULES_REVOKE = "/{roleId}/modules/{moduleId}";
    
    // ═══════════════════════════════════════════════════════════════
    // FILE UPLOAD
    // ═══════════════════════════════════════════════════════════════
    
    public static final String FILES_BASE = API_BASE;
    public static final String FILES_UPLOAD = "/upload";
    public static final String FILES_DOWNLOAD = "/files/{filename}";
    
    // ═══════════════════════════════════════════════════════════════
    // LEVEL UP (User Permission Elevation)
    // ═══════════════════════════════════════════════════════════════
    
    public static final String LEVEL_UP_BASE = API_BASE;
    public static final String LEVEL_UP_REQUEST = "/level-up/request";
    public static final String LEVEL_UP_PENDING = "/level-up/pending";
    public static final String LEVEL_UP_DECIDE = "/level-up/{id}/decide";
    
    // ═══════════════════════════════════════════════════════════════
    // RESPONSE CODES (HTTP Status Messages)
    // ═══════════════════════════════════════════════════════════════
    
    public static final class Messages {
        private Messages() {}
        
        // Success Messages
        public static final String LOGIN_SUCCESS = "Login successful";
        public static final String LOGOUT_SUCCESS = "Logout successful";
        public static final String REGISTER_SUCCESS = "User registered successfully";
        public static final String FORM_SAVED = "Form saved successfully";
        public static final String FORM_PUBLISHED = "Form published successfully";
        public static final String SUBMISSION_SUCCESS = "Submission successful";
        public static final String UPDATE_SUCCESS = "Update successful";
        public static final String DELETE_SUCCESS = "Deleted successfully";
        public static final String VERSION_ACTIVATED = "Version activated";
        
        // Error Messages
        public static final String INVALID_CREDENTIALS = "Invalid username or password";
        public static final String USER_EXISTS = "Username already exists";
        public static final String FORM_NOT_FOUND = "Form not found";
        public static final String UNAUTHORIZED = "Unauthorized access";
        public static final String VALIDATION_ERROR = "Validation failed";
    }
    
    // ═══════════════════════════════════════════════════════════════
    // ERROR CODES (Custom Application Errors)
    // ═══════════════════════════════════════════════════════════════
    
    public static final class ErrorCodes {
        private ErrorCodes() {}
        
        public static final String FORM_NOT_FOUND = "FORM_NOT_FOUND";
        public static final String DRAFT_NOT_FOUND = "DRAFT_NOT_FOUND";
        public static final String FORM_ARCHIVED = "FORM_ARCHIVED";
        public static final String DRAFT_DISCARDED = "DRAFT_DISCARDED";
        public static final String UNAUTHORIZED = "UNAUTHORIZED";
        public static final String FORBIDDEN = "FORBIDDEN";
        public static final String FORM_NOT_PUBLISHED = "FORM_NOT_PUBLISHED";
        public static final String DUPLICATE_FORM_CODE = "DUPLICATE_FORM_CODE";
        public static final String VERSION_MISMATCH = "VERSION_MISMATCH";
        public static final String ALREADY_ACTIVE = "ALREADY_ACTIVE";
        public static final String CONCURRENT_SUBMISSION_REJECTED = "CONCURRENT_SUBMISSION_REJECTED";
        public static final String SCHEMA_DRIFT_DETECTED = "SCHEMA_DRIFT_DETECTED";
        public static final String SQL_RESERVED_KEYWORD = "SQL_RESERVED_KEYWORD";
        public static final String TYPE_MISMATCH = "TYPE_MISMATCH";
        public static final String INVALID_FIELD_KEY = "INVALID_FIELD_KEY";
        public static final String TYPE_STABILITY_VIOLATED = "TYPE_STABILITY_VIOLATED";
        public static final String INVALID_FORM_CODE = "INVALID_FORM_CODE";
        public static final String VALIDATION_ERROR = "VALIDATION_ERROR";
        public static final String BAD_REQUEST = "BAD_REQUEST";
    }
}

File: formbuilder-frontend1/src/utils/apiConstants.ts
/**
 * API Constants
 * 
 * Centralized API endpoint paths and configuration.
 * Single source of truth for all API calls in the frontend.
 * 
 * When API version changes, update VERSION constant to update all endpoints.
 */

// ═══════════════════════════════════════════════════════════════
// BASE CONFIGURATION
// ═══════════════════════════════════════════════════════════════

/** Backend server URL - Change for different environments */
export const API_SERVER = 'http://localhost:8080';

/** Current API Version - Change this to update all endpoints */
export const API_VERSION = 'v1';

/** Base API Path */
export const API_BASE = `${API_SERVER}/api/${API_VERSION}`;

// ═══════════════════════════════════════════════════════════════
// AUTHENTICATION & SESSION
// ═══════════════════════════════════════════════════════════════

export const AUTH = {
  BASE: `${API_BASE}/auth`,
  LOGIN: `${API_BASE}/auth/login`,
  LOGOUT: `${API_BASE}/auth/logout`,
  REGISTER: `${API_BASE}/auth/register`,
  ME: `${API_BASE}/auth/me`,
  PERMISSIONS: `${API_BASE}/auth/permissions`,
} as const;

// ═══════════════════════════════════════════════════════════════
// FORM MANAGEMENT
// ═══════════════════════════════════════════════════════════════

export const FORMS = {
  BASE: `${API_BASE}/forms`,
  LIST: `${API_BASE}/forms`,
  CREATE: `${API_BASE}/forms`,
  GET: (id: string | number) => `${API_BASE}/forms/${id}`,
  UPDATE: (id: string | number) => `${API_BASE}/forms/${id}`,
  DELETE: (id: string | number) => `${API_BASE}/forms/${id}`,
  RESTORE: (id: string | number) => `${API_BASE}/forms/${id}/restore`,
  DELETE_PERMANENT: (id: string | number) => `${API_BASE}/forms/${id}/permanent`,
  ARCHIVED: `${API_BASE}/forms/archived`,
  STATS: `${API_BASE}/forms/stats`,
  
  // Public form access (no auth required)
  PUBLIC: (token: string) => `${API_BASE}/forms/public/${token}`,
  PUBLIC_SUBMISSIONS: (token: string) => `${API_BASE}/forms/public/${token}/submissions`,
  PUBLIC_SUBMISSION_GET: (token: string, submissionId: string) => 
    `${API_BASE}/forms/public/${token}/submissions/${submissionId}`,
  PUBLIC_SUBMISSION_UPDATE: (token: string, submissionId: string) => 
    `${API_BASE}/forms/public/${token}/submissions/${submissionId}`,
} as const;

// ═══════════════════════════════════════════════════════════════
// FORM VERSIONS
// ═══════════════════════════════════════════════════════════════

export const VERSIONS = {
  BASE: (formId: string | number) => `${API_BASE}/forms/${formId}/versions`,
  CREATE: (formId: string | number) => `${API_BASE}/forms/${formId}/versions`,
  LIST: (formId: string | number) => `${API_BASE}/forms/${formId}/versions`,
  GET: (formId: string | number, versionId: string | number) => `${API_BASE}/forms/${formId}/versions/${versionId}`,
  ACTIVATE: (formId: string | number, versionId: string | number) => `${API_BASE}/forms/${formId}/versions/${versionId}/activate`,
} as const;

// ═══════════════════════════════════════════════════════════════
// SUBMISSIONS
// ═══════════════════════════════════════════════════════════════

export const SUBMISSIONS = {
  LIST: (formId: string | number) => `${API_BASE}/forms/${formId}/submissions`,
  CREATE: (formId: string | number) => `${API_BASE}/forms/${formId}/submissions`,
  GET: (formId: string | number, submissionId: string | number) => `${API_BASE}/forms/${formId}/submissions/${submissionId}`,
  UPDATE: (formId: string | number, submissionId: string | number) => `${API_BASE}/forms/${formId}/submissions/${submissionId}`,
  DELETE: (formId: string | number, submissionId: string | number) => `${API_BASE}/forms/${formId}/submissions/${submissionId}`,
  BULK: (formId: string | number) => `${API_BASE}/forms/${formId}/submissions/bulk`,
  RESTORE: (formId: string | number, submissionId: string | number) => `${API_BASE}/forms/${formId}/submissions/${submissionId}/restore`,
  EXPORT: (formId: string | number) => `${API_BASE}/forms/${formId}/submissions/export`,
  
  // Lookup values for dropdowns
  LOOKUP_VALUES: (formId: string, fieldKey: string) => 
    `${API_BASE}/forms/${formId}/columns/${fieldKey}/values`,
} as const;

// ═══════════════════════════════════════════════════════════════
// RUNTIME (Form Fill)
// ═══════════════════════════════════════════════════════════════

export const RUNTIME = {
  BASE: `${API_BASE}/runtime`,
  GET_FORM: (formCode: string) => `${API_BASE}/runtime/forms/${formCode}`,
  SUBMIT: (formCode: string) => `${API_BASE}/runtime/forms/${formCode}/submissions`,
  SAVE_DRAFT: (formCode: string) => `${API_BASE}/runtime/forms/${formCode}/drafts`,
} as const;

// ═══════════════════════════════════════════════════════════════
// WORKFLOW (Approval System)
// ═══════════════════════════════════════════════════════════════

export const WORKFLOW = {
  BASE: `${API_BASE}/workflows`,
  INITIATE: `${API_BASE}/workflows/initiate`,
  APPROVE: (stepId: string) => `${API_BASE}/workflows/steps/${stepId}/approve`,
  REJECT: (stepId: string) => `${API_BASE}/workflows/steps/${stepId}/reject`,
  AUTHORITIES: `${API_BASE}/workflows/available-authorities`,
  MY_SUBMISSIONS: `${API_BASE}/workflows/my-submissions`,
  MY_PENDING: `${API_BASE}/workflows/my-pending`,
  MY_HANDLED: `${API_BASE}/workflows/my-handled`,
  FIX_DB: `${API_BASE}/workflows/fix-db`,
} as const;

// ═══════════════════════════════════════════════════════════════
// ADMIN - USER MANAGEMENT
// ═══════════════════════════════════════════════════════════════

export const ADMIN_USERS = {
  BASE: `${API_BASE}/admin/users`,
  LIST: `${API_BASE}/admin/users`,
  CREATE: `${API_BASE}/admin/users`,
  GET: (id: string | number) => `${API_BASE}/admin/users/${id}`,
  UPDATE: (id: string | number) => `${API_BASE}/admin/users/${id}`,
  DELETE: (id: string | number) => `${API_BASE}/admin/users/${id}`,
  TOGGLE_DELETE: (id: string | number) => `${API_BASE}/admin/users/${id}/toggle-delete`,
  ASSIGN_ROLE: (userId: string | number) => `${API_BASE}/admin/users/${userId}/roles`,
  REVOKE_ROLE: (userId: string | number, assignmentId: string | number) => 
    `${API_BASE}/admin/users/${userId}/roles/${assignmentId}`,
} as const;

// ═══════════════════════════════════════════════════════════════
// ADMIN - ROLE MANAGEMENT
// ═══════════════════════════════════════════════════════════════

export const ADMIN_ROLES = {
  BASE: `${API_BASE}/admin/roles`,
  LIST: `${API_BASE}/admin/roles`,
  CREATE: `${API_BASE}/admin/roles`,
  GET: (id: string) => `${API_BASE}/admin/roles/${id}`,
  UPDATE: (id: string) => `${API_BASE}/admin/roles/${id}`,
  DELETE: (id: string) => `${API_BASE}/admin/roles/${id}`,
} as const;

// ═══════════════════════════════════════════════════════════════
// ADMIN - PERMISSIONS
// ═══════════════════════════════════════════════════════════════

export const ADMIN_PERMISSIONS = {
  BASE: `${API_BASE}/admin/permissions`,
  LIST: `${API_BASE}/admin/permissions`,
} as const;

// ═══════════════════════════════════════════════════════════════
// ADMIN - AUDIT LOGS
// ═══════════════════════════════════════════════════════════════

export const ADMIN_AUDIT = {
  BASE: `${API_BASE}/admin/audit`,
  LIST: `${API_BASE}/admin/audit`,
} as const;

// ═══════════════════════════════════════════════════════════════
// ADMIN - APPROVAL HISTORY
// ═══════════════════════════════════════════════════════════════

export const ADMIN_APPROVALS = {
  BASE: `${API_BASE}/admin/approvals/history`,
  LIST: `${API_BASE}/admin/approvals/history`,
} as const;

// ═══════════════════════════════════════════════════════════════
// USER PROFILE
// ═══════════════════════════════════════════════════════════════

export const PROFILE = {
  BASE: `${API_BASE}/profile`,
  UPDATE: `${API_BASE}/profile`,
  CHANGE_PASSWORD: `${API_BASE}/profile/change-password`,
} as const;

// ═══════════════════════════════════════════════════════════════
// MODULES & MENU (Dynamic Navigation)
// ═══════════════════════════════════════════════════════════════

export const MODULES = {
  BASE: `${API_BASE}/modules`,
  LIST: `${API_BASE}/modules`,
} as const;

export const MENU = {
  BASE: `${API_BASE}/menu`,
  LIST: `${API_BASE}/menu`,
} as const;

// ═══════════════════════════════════════════════════════════════
// ROLE-MODULE MAPPING
// ═══════════════════════════════════════════════════════════════

export const ROLE_MODULES = {
  BASE: `${API_BASE}/roles`,
  GET: (roleId: string) => `${API_BASE}/roles/${roleId}/modules`,
  ASSIGN: (roleId: string) => `${API_BASE}/roles/${roleId}/modules`,
  REVOKE: (roleId: string, moduleId: string) => `${API_BASE}/roles/${roleId}/modules/${moduleId}`,
} as const;

// ═══════════════════════════════════════════════════════════════
// FILE UPLOAD
// ═══════════════════════════════════════════════════════════════

export const FILES = {
  UPLOAD: `${API_BASE}/upload`,
  DOWNLOAD: (filename: string) => `${API_BASE}/files/${filename}`,
} as const;

// ═══════════════════════════════════════════════════════════════
// LEVEL UP (User Permission Elevation)
// ═══════════════════════════════════════════════════════════════

export const LEVEL_UP = {
  REQUEST: `${API_BASE}/level-up/request`,
  PENDING: `${API_BASE}/level-up/pending`,
  DECIDE: (id: string) => `${API_BASE}/level-up/${id}/decide`,
} as const;

// ═══════════════════════════════════════════════════════════════
// HELPER FUNCTIONS
// ═══════════════════════════════════════════════════════════════

/**
 * Build URL with query parameters
 * @example
 * buildUrl(FORMS.LIST, { page: 0, size: 10 })
 * // Returns: "http://localhost:8080/api/v1/forms?page=0&size=10"
 */
export function buildUrl(baseUrl: string, params?: Record<string, string | number | boolean | null | undefined>): string {
  if (!params || Object.keys(params).length === 0) {
    return baseUrl;
  }
  
  const queryString = new URLSearchParams(
    Object.entries(params)
      .filter(([, value]) => value !== undefined && value !== null)
      .map(([key, value]) => [key, String(value)])
  ).toString();
  
  return queryString ? `${baseUrl}?${queryString}` : baseUrl;
}

/**
 * Standard fetch options with credentials
 */
export const fetchOptions = {
  credentials: 'include' as RequestCredentials,
};

/**
 * GET request helper
 */
export function getOptions(): RequestInit {
  return {
    method: 'GET',
    ...fetchOptions,
  };
}

/**
 * POST request helper
 */
export function postOptions(body?: unknown): RequestInit {
  return {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: body ? JSON.stringify(body) : undefined,
    ...fetchOptions,
  };
}

/**
 * PUT request helper
 */
export function putOptions(body?: unknown): RequestInit {
  return {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: body ? JSON.stringify(body) : undefined,
    ...fetchOptions,
  };
}

/**
 * DELETE request helper
 */
export function deleteOptions(): RequestInit {
  return {
    method: 'DELETE',
    ...fetchOptions,
  };
}

// ═══════════════════════════════════════════════════════════════
// EXPORT ALL AS DEFAULT FOR CONVENIENCE
// ═══════════════════════════════════════════════════════════════

const apiConstants = {
  AUTH,
  FORMS,
  VERSIONS,
  SUBMISSIONS,
  RUNTIME,
  WORKFLOW,
  ADMIN_USERS,
  ADMIN_ROLES,
  ADMIN_PERMISSIONS,
  ADMIN_AUDIT,
  ADMIN_APPROVALS,
  PROFILE,
  MODULES,
  MENU,
  ROLE_MODULES,
  FILES,
  LEVEL_UP,
  buildUrl,
  fetchOptions,
  getOptions,
  postOptions,
  putOptions,
  deleteOptions,
};

export default apiConstants;
